google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java

```diff
@@ -0,0 +1,263 @@
+      // ListPaymentsAccounts method.
+      billingSetupBuilder.setPaymentsAccount(
+          StringValue.of(ResourceNames.paymentsAccount(customerId, paymentsAccountId)));
+    } else if (paymentsProfileId != null) {
+      // Otherwise, create a new payments account by setting the PaymentsAccountInfo
+      // field. See https://support.google.com/google-ads/answer/7268503 for information
+      // about payments profiles.
+      billingSetupBuilder.setPaymentsAccountInfo(
+          PaymentsAccountInfo.newBuilder()
+              .setPaymentsAccountName(
+                  StringValue.of("Payments Account #" + System.currentTimeMillis()))
+              .setPaymentsProfileId(StringValue.of(paymentsProfileId))
+              .build());
+    } else {
+      throw new Exception("No paymentsAccountId or paymentsProfileId provided.");
+    }
+
+    return billingSetupBuilder.build();
+  }
+
+  /**
+   * Sets the starting and ending date times for the new billing setup. Queries the customer's
+   * account to see if there are any approved billing setups. If there are any, the new billing
+   * setup starting date time is set to one day after the last. If not, the billing setup is set to
+   * start immediately. The ending date is set to one day after the starting date time.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the customer ID containing the BillingSetup to remove.
+   * @param billingSetup The instance of BillingSetup whose starting date time will be set.
+   * @return Billing Setup with updated start and end date time.
+   */
+  private BillingSetup setBillingSetupDateTimes(
+      GoogleAdsClient googleAdsClient, long customerId, BillingSetup billingSetup)
+      throws Exception {
+    // The query to search existing approved billing setups in the end date time descending
+    // order.
+    // See GetBillingSetup.java for a more detailed example of requesting billing setup
+    // information.
+    String query =
+        "SELECT billing_setup.end_date_time "
+            + "FROM billing_setup "
+            + "WHERE billing_setup.status = 'APPROVED' "
+            + "ORDER BY billing_setup.end_date_time DESC";
+
+    // Creates the service client.
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      // Issues a search request.
+      SearchPagedResponse searchPagedResponse =
+          googleAdsServiceClient.search(Long.toString(customerId), query);
+
+      BillingSetup.Builder billingSetupBuilder = billingSetup.toBuilder();
+      if (searchPagedResponse.getPage().getResponse().getResultsCount() > 0) {
+        // Retrieves the ending date time of the last billing setup.
+        StringValue lastEndingDateTimeString =
+            searchPagedResponse
+                .getPage()
+                .getResponse()
+                .getResults(0)
```

Comment:
Would it be clearer to do searchPagedResponse.iterator().next()?

Reply:
Iterator is not available on `searchPagedResponse`. You could do `searchPagedResponse.iterateAll().iterator().next()` but I think that's actually more confusing.